src/main/java/com/schedule/workout/workoutScheduler/services/UserRolesService.java

```diff
@@ -0,0 +1,116 @@
+package com.schedule.workout.workoutScheduler.services;
+
+import com.schedule.workout.workoutScheduler.controller.userRole.CreateUserRoleModel;
+import com.schedule.workout.workoutScheduler.controller.userRole.UpdateUserRoleModel;
+import com.schedule.workout.workoutScheduler.controller.userRole.UserRoleModel;
+import com.schedule.workout.workoutScheduler.database.IUsersRepository;
+import com.schedule.workout.workoutScheduler.database.IRolesRepository;
+import com.schedule.workout.workoutScheduler.database.IUserRoleRepository;
+import com.schedule.workout.workoutScheduler.database.model.RoleDB;
+import com.schedule.workout.workoutScheduler.database.model.UserDB;
+import com.schedule.workout.workoutScheduler.database.model.UserRoleDB;
+import com.schedule.workout.workoutScheduler.exceptions.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class UserRolesService {
+
+    @Autowired
+    IUsersRepository usersRepository;
+    @Autowired
+    IRolesRepository rolesRepository;
+    @Autowired
+    IUserRoleRepository userRoleRepository;
+
+    public CreateUserRoleModel createUserRole(CreateUserRoleModel createUserRoleModel) {
+        List<UserRoleDB> userRoles = userRoleRepository.findByUserIdOrRoleId(createUserRoleModel.getUserID(),createUserRoleModel.getRoleID());
```

Comment:
Can you just check if there is a record with specific combo of user id and role id. If there is it means user is already linked to that role. No need for this complex logic. 

Also instead of using nested ifs and throwing exceptions in else, its more concise to fail fast. 

For example:

if (userDB == null) throw new UserNotFoundException();
if (roleDB == null) throw new RoleNotFoundException();

if this passes it means all validations are good so you can just continue with happy flow